### ISSUE TITLE
[Icon]: Interactive Icon Does not launch when we hit "Enter" / "Space" with Screen readers likes JAWS/NVDA

### ISSUE THREAD

**Issue by achuth2912** [SAP/ui5-webcomponents#12788]
### Describe the bug

When we navigate with the Screen readers like JAWS/NVDA and use the Icon component with interactive mode the Icon does not work as expected when we hit "Enter"/"Space" i.e. via the keyboard.

This works fine with out any Screen reader. This issue happens only when we have the JAWS/NVDA and try to interact with the Keyboard.

### Isolated Example

https://stackblitz.com/edit/github-xfwxnan5?file=src%2FApp.tsx

### Reproduction steps

1. Turn on any Screen reader like JAWS /NVDA
2. Use Keyboard to navigate to the icon and then try "Enter" / "Space" it does not work. 

<img width="1726" height="925" alt="Image" src="https://github.com/user-attachments/assets/fa1e6fc5-e728-4fed-a03a-63dcb8c23dd8" />

...

### Expected Behaviour

Even when we use the Screen readers like JAWS/NVDA the Interactive Icon must work with Keyboard hit like "Enter"/"Space"

### UI5 Web Components for React Version

2.17.0

### UI5 Web Components Version

2.17.0

### Browser

Chrome

### Operating System

Windows

### Relevant log output

```shell

```

### Organization

SAP

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by MarcusNotheis** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.

**Comment by didip1000** [comment]
Hi @achuth2912,

Can you confirm if JAWS/NVDA is in VPC mode? 

Thanks,
Diana

**Comment by didip1000** [comment]
Hey @UI5/ui5-team-ctr-pir,

I could not reproduce the issue on JAWS in Interactive mode, only in VPC mode which could be related to [this known JAWS issue](https://github.com/FreedomScientific/standards-support/issues/899). 

However the issue is reproducible with NVDA, and it's also reproducible in [this native HTML sample](https://jsfiddle.net/n0yqt45b/). It seems like with NVDA only `click` events are sent.

Can you please take a look into why?

Thanks,
Diana

**Comment by achuth2912** [comment]
Any Update on this Issue ?
We have a customer Incident waiting on this one

**Comment by PetyaMarkovaBogdanova** [comment]
Hi @achuth2912 ,
unfortunately, as @didip1000 has suggested in her comment above, there is an open JAWS [issue](https://github.com/FreedomScientific/standards-support/issues/899) about suppressing keyboard events when reader is open, and it happens with both NVDA and JAWS, it is related to automatically setting the VPC mode to "on". They are only firing click event on "Enter", which is not the case in ui5-icon, because we do not recommend using interactive icon as a press handler, as per UX Guidelines "If interaction is required, always display the icon in an appropriate container and apply the interaction to this container component." So as a workaround you can use the icon only button.

Related issues: https://github.com/UI5/webcomponents/issues/11980

Best regards,
Petya Markova.
UI5 Web Components Dispatcher
